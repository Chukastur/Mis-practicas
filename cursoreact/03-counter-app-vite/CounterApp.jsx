import { useState } from "react";


export const CounterApp = ( { value = 0 } ) => {    

    
    const [ counter, setCounter ] = useState ( value )

    const handleAdd = () => {        
        setCounter ( counter + 1 );
        // setCounter ( (c) => c + 1 )
    }

    const handleSubstract = () => {        
        setCounter ( counter - 1 );
    }

    const handleReset = () => {        
        setCounter ( value );
    }



    if (typeof value !=  'number' ) {
    console.error("El prop 'value' debe ser un n√∫mero.");
    }  

    return (
        <>
        <h1>CounterApp</h1>
        <p>{ counter }</p>

        <button onClick = { handleAdd }> +1 </button>
        <button onClick = { handleSubstract }> -1 </button>
        <button aria-label = "btn-reset" onClick = { handleReset }> Reset </button>

        </>
    );

};

    