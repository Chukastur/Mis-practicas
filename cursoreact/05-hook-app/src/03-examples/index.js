

export * from './MultipleCustomHooks'
export * from './LoadingMesage'
export * from './PokemonCard'
export * from '../hooks/useCounter'
export * from '../hooks/useFetch'
export * from '../hooks/useForm'

