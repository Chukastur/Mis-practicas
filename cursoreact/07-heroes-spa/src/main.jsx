import { StrictMode, react } from 'react';
import { createRoot } from 'react-dom/client';
import { BrowserRouter, RouterProvider } from "react-router-dom";

import {HeroesApp} from './HeroesApp';


createRoot(document.getElementById('root')).render(
  <StrictMode>
    
   <BrowserRouter>
    <HeroesApp />
  </BrowserRouter>
   
  </StrictMode>
);
