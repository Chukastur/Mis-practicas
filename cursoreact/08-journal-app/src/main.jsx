
import { BrowserRouter } from 'react-router-dom';
import { createRoot } from 'react-dom/client'
import { Provider } from 'react-redux'
import { StrictMode } from 'react'

import { store } from './store'
import { JournalApp } from './JournalApp'
import './styles.css'

createRoot(document.getElementById('root')).render(

  <StrictMode>
    <Provider store={store}>
      <BrowserRouter>
        <JournalApp />
      </BrowserRouter>
    </Provider>
  </StrictMode>

);
