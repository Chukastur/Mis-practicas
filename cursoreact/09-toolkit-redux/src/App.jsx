import { useDispatch, useSelector } from 'react-redux';
import { decrement, increment, incrementBy } from './store';
import './App.css';

function App() {

  const { counter } = useSelector ( state => state.counter);
  const dispatch = useDispatch();

  return (
    <>
      <h1>Vite + React</h1>
      <p>count is { counter }</p>
      <div>
        <button onClick={ () => dispatch ( increment () ) }>Suma</button>
          
        <button onClick={ () => dispatch ( decrement () ) }>Resta</button>

        <button onClick={ () => dispatch ( incrementBy (2) ) }>Suma 2</button> 
      </div>
    </>

  );
};

export default App;
