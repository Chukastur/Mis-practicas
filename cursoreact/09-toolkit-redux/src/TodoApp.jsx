import { useState } from "react";
import { useGetTodoByIdQuery, useGetTodosQuery } from "./store/apis/todosApi";


export const TodoApp = () => {

  const [todoId, setTodoId ] = useState (1);
                                                 //destructuramos y renombramos data
  //const { data: todos = [] , isLoading } = useGetTodosQuery ();
  const { data: todo, isLoading } = useGetTodoByIdQuery (todoId); 
  
  const nextTodo = () => { 
    setTodoId ( todoId + 1 );
  };

  const prevTodo = () => { 
    if ( todoId === 1) return;
    setTodoId ( todoId - 1 );
  };
  

  return (
   <>
      <h1>Todos - RTK Query</h1>
      <hr />
      <h4>isLoading...{ isLoading ? 'Tue' : 'False'}</h4>

      <pre>{ JSON.stringify ( todo )}</pre>

      <button onClick={ prevTodo }>
        Prev Todo
      </button>
      <button onClick={ nextTodo }>
        Next Todo
      </button>
   </>
  );
};
