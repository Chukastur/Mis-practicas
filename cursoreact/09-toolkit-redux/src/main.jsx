import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { Provider } from 'react-redux';
import { store } from './';
// import App from './App.jsx';
import { PokemonApp } from './PokemonApp.jsx';
import './index.css';
import { TodoApp } from './TodoApp.jsx';



createRoot(document.getElementById('root')).render(
  <StrictMode>

    <Provider store={ store }>
      
      {/* <App />       */}
      {/* <PokemonApp /> */}
      {<TodoApp />}

    </Provider>

  </StrictMode>
);
  