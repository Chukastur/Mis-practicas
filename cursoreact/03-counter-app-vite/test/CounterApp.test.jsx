import { CounterApp } from "../src/CounterApp";
import { render, screen, fireEvent } from '@testing-library/react';



describe('Pruebas en CounterApp', () => { 

    const initialValue = 10;

    test('debe hacer match con el SnapShot', () => {

        const { container } = render ( <CounterApp value = { initialValue } /> );
        expect ( container ).toMatchSnapshot ();
        
    });

    test('debe mostrar el valor inicial de 100', () => {

        render ( <CounterApp value = { 100 } /> ); 
        expect ( screen.getByText( 100 ).innerHTML ).toBe ('100');
        
    });

    test('debe incrementar con el botón +1', () => {
        
        render ( <CounterApp value = { initialValue } /> );
        fireEvent.click ( screen.getByText ( '+1' ) );
        expect ( screen.getByText ( '11' ) ).toBeTruthy ();

    });

    test('debe decrementar con el botón -1', () => {
        
        render ( <CounterApp value = { initialValue } /> );
        fireEvent.click ( screen.getByText ( '-1' ) );
        expect ( screen.getByText ( '9' ) ).toBeTruthy ();

    });

    test('debe funcionar el botón reset', () => {

        render ( <CounterApp value = { initialValue } /> );
        fireEvent.click ( screen.getByText ( '+1' ) );
        fireEvent.click ( screen.getByText ( '+1' ) );
        fireEvent.click ( screen.getByText ( '+1' ) );
        //fireEvent.click ( screen.getByText ( 'Reset' ) );
        fireEvent.click (screen.getByRole ('button', { name: 'btn-reset' }));


        expect ( screen.getByText ( '10' ) ).toBeTruthy () ;

    });


 });     