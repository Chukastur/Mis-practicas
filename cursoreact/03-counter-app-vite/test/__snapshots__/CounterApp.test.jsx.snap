// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en CounterApp debe hacer match con el SnapShot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
    10
  </p>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
